src/main/kotlin/org/rust/lang/core/stubs/StubInterfaces.kt

```diff
@@ -5,6 +5,55 @@
 
 package org.rust.lang.core.stubs
 
+import com.intellij.util.BitUtil
+import org.rust.lang.core.psi.ext.QueryAttributes
+import org.rust.lang.core.psi.ext.RsDocAndAttributeOwner
+import org.rust.lang.core.psi.ext.name
+import org.rust.lang.core.psi.ext.queryAttributes
+import org.rust.stdext.makeBitMask
+
 interface RsNamedStub {
     val name: String?
 }
+
+interface RsDocAndAttributeOwnerStub {
+    // These properties are stored in stubs for performance reasons: it's much cheaper to check
+    // a flag in a stub then traverse a PSI
+
+    val hasAttrs: Boolean
+
+    // #[cfg()]
+    val hasCfg: Boolean
```

Comment:
Probably, we need a slightly better name because of `cfg_attr` to mark these properties as approximate because we can't evaluate `cfg_attr` while indexing. Or, we need some documentation about it

Reply:
I think we can care about during `cfg_attr` implementation. There are at least 2 different ways how we can support then and chosen way can affect names & documentation

Reply:
okay